compute/instances/create-start-instance/create_instance_with_snapshotted_data_disk.go

```diff
@@ -0,0 +1,124 @@
+	// projectID := "your_project_id"
+	// zone := "europe-central2-b"
+	// instanceName := "your_instance_name"
+	// snapshotLink := "projects/project_name/global/snapshots/snapshot_name"
+
+	ctx := context.Background()
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewInstancesRESTClient: %v", err)
+	}
+	imagesClient, err := compute.NewImagesRESTClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewImagesRESTClient: %v", err)
+	}
+	defer instancesClient.Close()
+	defer imagesClient.Close()
+
+	// List of public operating system (OS) images: https://cloud.google.com/compute/docs/images/os-details.
+	newestDebianReq := &computepb.GetFromFamilyImageRequest{
+		Project: "debian-cloud",
+		Family:  "debian-10",
+	}
+	newestDebian, err := imagesClient.GetFromFamily(ctx, newestDebianReq)
+	if err != nil {
+		return fmt.Errorf("unable to get image from family: %v", err)
+	}
+
+	req := &computepb.InsertInstanceRequest{
+		Project: projectID,
+		Zone:    zone,
+		InstanceResource: &computepb.Instance{
+			Name: proto.String(instanceName),
+			Disks: []*computepb.AttachedDisk{
+				{
+					InitializeParams: &computepb.AttachedDiskInitializeParams{
+						DiskSizeGb:  proto.Int64(10),
+						SourceImage: newestDebian.SelfLink,
+						DiskType:    proto.String(fmt.Sprintf("zones/%s/diskTypes/pd-standard", zone)),
+					},
+					AutoDelete: proto.Bool(true),
+					Boot:       proto.Bool(true),
+					Type:       proto.String(computepb.AttachedDisk_PERSISTENT.String()),
+				},
+				{
+					InitializeParams: &computepb.AttachedDiskInitializeParams{
+						DiskSizeGb:     proto.Int64(11),
+						SourceSnapshot: proto.String(snapshotLink),
+						DiskType:       proto.String(fmt.Sprintf("zones/%s/diskTypes/pd-standard", zone)),
+					},
+					AutoDelete: proto.Bool(true),
+					Boot:       proto.Bool(false),
+					Type:       proto.String(computepb.AttachedDisk_PERSISTENT.String()),
+				},
+			},
+			MachineType: proto.String(fmt.Sprintf("zones/%s/machineTypes/n1-standard-1", zone)),
+			NetworkInterfaces: []*computepb.NetworkInterface{
+				{
+					Name: proto.String("global/networks/default"),
+				},
+			},
+		},
+	}
+
+	op, err := instancesClient.Insert(ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to create instance: %v", err)
+	}
+
+	zoneOperationsClient, err := compute.NewZoneOperationsRESTClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewZoneOperationsRESTClient: %v", err)
+	}
+	defer zoneOperationsClient.Close()
+
+	for {
+		waitReq := &computepb.WaitZoneOperationRequest{
+			Operation: op.Proto().GetName(),
+			Project:   projectID,
+			Zone:      zone,
+		}
+		zoneOp, err := zoneOperationsClient.Wait(ctx, waitReq)
+		if err != nil {
+			return fmt.Errorf("unable to wait for the operation: %v", err)
+		}
+
+		if *zoneOp.Status.Enum() == computepb.Operation_DONE {
+			fmt.Fprintf(w, "Instance created\n")
```

Comment:
Since there is no formatting used here you could use fmt.Fprintln instead and omit the newline.